[FR] Add stack_id and an optional print of stack_id to stack_trace mapping

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #160119

To better help users debug with FR, we want to add stack_id and print a map between stack_id and stack_trace (optional)

Screenshot:

<img width="1029" height="529" alt="image" src="https://github.com/user-attachments/assets/8404a1d3-cc33-4f5f-971b-29609ec316c1" />

<img width="1620" height="358" alt="image" src="https://github.com/user-attachments/assets/3dd29c8c-ff68-41a2-acfd-e770036cfeb1" />

fduwjj committed:
[FR] Add stack_id and an optional print of stack_id to stack_trace mapping

[ghstack-poisoned]

fduwjj committed:
Update on "[FR] Add stack_id and an optional print of stack_id to stack_trace mapping"


To better help users debug with FR, we want to add stack_id and print a map between stack_id and stack_trace (optional)

[ghstack-poisoned]

fduwjj committed:
Update on "[FR] Add stack_id and an optional print of stack_id to stack_trace mapping"


To better help users debug with FR, we want to add stack_id and print a map between stack_id and stack_trace (optional)

[ghstack-poisoned]

wconstab commented on tools/flight_recorder/components/utils.py:
```diff
@@ -677,6 +689,27 @@ def get_version_detail(version: str) -> tuple[int, int]:
+    for rank in entries:
+        for dump in entries[rank]:
+            if dump.get("frames", []):
+                frames = str(dump["frames"])
```
can you remind me how the frames are stored in the on-disk file?  iirc they are already stored separately from the entries and there is some pointer/id that the entry uses to look up the frames in the frames table right?

if so, it would be nice to directly use this info so that we do not need to hash the large string of frames every time we check if one frames is in the map already

wconstab reviewed (commented):
can you paste the new output in the PR desc?

H-Huang reviewed (approved):
Thanks for adding!

fduwjj committed:
Update on "[FR] Add stack_id and an optional print of stack_id to stack_trace mapping"


To better help users debug with FR, we want to add stack_id and print a map between stack_id and stack_trace (optional)

Screenshot:

<img width="1029" height="529" alt="image" src="https://github.com/user-attachments/assets/8404a1d3-cc33-4f5f-971b-29609ec316c1" />

<img width="1620" height="358" alt="image" src="https://github.com/user-attachments/assets/3dd29c8c-ff68-41a2-acfd-e770036cfeb1" />



[ghstack-poisoned]

fduwjj commented:
@pytorchbot  merge
